Sources/WalletConnectPush/Client/Wallet/WalletPushClient.swift

```diff
@@ -124,6 +126,23 @@ public class WalletPushClient {
     }
 }
 
+private extension WalletPushClient {
+
+    func registerIfNeeded(account: Account, onSign: @escaping SigningCallback) async throws {
+        guard !syncClient.isRegistered(account: account) else { return }
+
+        let result = await onSign(syncClient.getMessage(account: account))
+
+        switch result {
+        case .signed(let signature):
+            try await syncClient.register(account: account, signature: signature)
+            logger.debug("Sync pushSubscriptions store registered and initialized")
+        case .rejected:
+            throw PushError.registerSignatureRejected
+        }
+    }
+}
+
```

Comment:
could be move it to a separate class to not pollute wallet client?
syncRegisterService or smth like that

Reply:
created PushSyncService